framework/src/main/java/org/apache/drill/test/framework/DrillTestJdbc.java

```diff
@@ -300,6 +301,13 @@ private boolean isNullabilityCompatible(List<Integer> limitZero, List<Integer> r
   
   @Override
   public void cancel() {
+	if (statement != null)
+		try {
+			statement.cancel();
+		} catch (SQLException e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
+		}
     thread.interrupt();
```

Comment:
If the query was cancelled successfully, the `thread` will be unblocked in near future. So this `interrupt()` call could "hide" incorrect handling of cancellation.